File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/interactive/InteractiveUseEndedMessage.scala





















// Generated on 05/03/2013 10:34:52
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait InteractiveUseEndedMessage extends MessageDeserializer {
  val messageId = 6112

  protected trait Message extends BaseMessage with Serializable {
    val messageId = InteractiveUseEndedMessage.messageId

    val elemId: Int
    val skillId: Short

    def serialize(buf: Buffer) {
      buf.writeInt(elemId)
      buf.writeShort(skillId)
    }

  }

  protected def create(elemId: Int, skillId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val elemId = buf.readInt()
    val skillId = buf.readShort()

    create(elemId, skillId)
  }

}

object InteractiveUseEndedMessage extends InteractiveUseEndedMessage {
  case class MessageImpl(elemId: Int, skillId: Short) extends Message
  type Target = MessageImpl

  override def create(elemId: Int, skillId: Short) = MessageImpl(elemId, skillId)

  def apply(elemId: Int, skillId: Short) = create(elemId, skillId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/InteractiveUseEndedMessage.java


// Generated on 05/08/2013 19:37:52
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class InteractiveUseEndedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6112;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int elemId;
    public short skillId;
    
    public InteractiveUseEndedMessage() { }
    
    public InteractiveUseEndedMessage(int elemId, short skillId) {
        this.elemId = elemId;
        this.skillId = skillId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(elemId);
        buf.writeShort(skillId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        elemId = buf.readInt();
        if (elemId < 0)
            throw new RuntimeException("Forbidden value on elemId = " + elemId + ", it doesn't respect the following condition : elemId < 0");
        skillId = buf.readShort();
        if (skillId < 0)
            throw new RuntimeException("Forbidden value on skillId = " + skillId + ", it doesn't respect the following condition : skillId < 0");
    }
    
}
